src/Resolve/graphtype.jl

```diff
@@ -241,21 +241,26 @@ mutable struct Graph
             uuid_to_name::Dict{UUID,String},
             reqs::Requires,
             fixed::Dict{UUID,Fixed},
-            verbose::Bool = false
+            verbose::Bool = false,
+            julia_version::Union{VersionNumber,Nothing} = VERSION,
         )
         # make sure all versions of all packages know about julia uuid
         for (uuid,vnmap) in deps, vn in versions[uuid]
             get!(valtype(vnmap), vnmap, vn)["julia"] = uuid_julia
         end
 
         # Tell the resolver about julia itself
-        fixed[uuid_julia] = Fixed(VERSION)
         uuid_to_name[uuid_julia] = "julia"
-        versions[uuid_julia] = Set([VERSION])
-        deps[uuid_julia] = DepsValDict(VERSION => valtype(DepsValDict)())
-        compat[uuid_julia] = CompatValDict(VERSION => valtype(CompatValDict)())
+        if julia_version !== nothing
+            fixed[uuid_julia] = Fixed(julia_version)
+            versions[uuid_julia] = Set([julia_version])
+            deps[uuid_julia] = DepsValDict(julia_version => valtype(DepsValDict)())
+            compat[uuid_julia] = CompatValDict(julia_version => valtype(CompatValDict)())
+        else
+            versions[uuid_julia] = Set([])
```

Comment:
I don't really understand how this works. If we say there are no versions of Julia available, why isn't the resolver complaining about that when it resolves packages that have a dependency on Julia?

Reply:
TBH I don't know.  I wouldn't have done this at all, except that it broke when I didn't do this, so I initialized it to the empty set and it works.